README.md

```diff
@@ -1 +1,13 @@
-# factorial
\ No newline at end of file
+# Factorial
+Compute the factorial of a given input non-negative integer.
+
+## Exercise
+* Design and implement a method that computes factorial of the input non-negative integer and returns it.
+* Factorial of a number, 'n' is represented as *n!* and is defined as follows:
+   - *0!* = *1*
+   - *n!* = *n * (n-1)!* if *n* > 0
+   -      = *n * (n-1) * (n-2) * ... * 1*
+* For example, 3! = 3 * 2 * 1 = 6. 6! = 720.
+* Share and explain the time and space complexities for your solution.
+* If you think of multiple approaches to solve the problem, implement the solution which minimizes space complexity. Explain the other approaches, and your decision in comments above the code.
+**Note**: Do not use Ruby provided functionality.
```

Comment:
We should have an extra blank line above this so that the note goes into its own paragraph.

Reply:
Good point! It was a copy-paste remnant from splitting up the exercises. I got rid of the note for this exercise.